### PR TITLE
Introduce reduce_cluster_test

### PR DESCRIPTION
**Pull request by asias**
Can be used to shrink a cluster size from X nodes to Y nodes.

### CODE REVIEW

File: reduce_cluster_test.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        self._cluster_starting_size = cluster_starting_size
+        self._cluster_target_size = cluster_target_size
+        self.init_resources(n_db_nodes=self._cluster_starting_size, n_loader_nodes=1)
+        self.loaders.wait_for_init()
+        self.db_cluster.wait_for_init()
+
+    def reduce_cluster(self, cluster_starting_size, cluster_target_size=3):
+        self.wait_for_init(cluster_starting_size, cluster_target_size)
+        nodes_to_remove = self._cluster_starting_size - self._cluster_target_size
+        # FIXME: we should find a way to stop c-s when decommission is finished instead of estimating time to run
```
Comment (lmr) [inline reduce_cluster_test.py]: Maybe we could send a SIGTERM to c-s. I still haven't seen what is the behavior in that case. If SIGTERM prompts c-s to end gracefully, that would be our solution.

File: sdcm/cluster.py [hunk 2]
```diff
@@ -691,16 +691,24 @@ def add_nemesis(self, nemesis):
                                     termination_event=self.termination_event))
 
     def start_nemesis(self, interval=30):
+        self.log.debug('Start nemesis beg')
```
Comment (lmr) [inline sdcm/cluster.py]: The message could ready just 'Start nemesis'.

Comment (lmr) [inline sdcm/cluster.py]: Or 'Start nemesis begin'. It's better than shortening the word 'begin'.

**Comment by lmr** [comment]
This looks good for the most part. I'll test it and then merge it.

File: sdcm/cluster.py (hunk 2 above)
Comment (asias) [inline sdcm/cluster.py]: OK. I will change.

**Comment by lmr** [comment]
Nice, I got decommission to fail here a bunch, seems we got a new great test. Merged, thanks @asias !

File: reduce_cluster_test.py (hunk 1 above)
Comment (asias) [inline reduce_cluster_test.py]: I checked that c-s will generate two processes, one for the script process, one for the java process. We need to kill them both. 

I tried:

pkill -TERM -P 9640

9640 is the parent process.

It seems it is working.

**Pull request merged**